packages/pwa-kit-react-sdk/src/ssr/universal/components/refetch-data/index.jsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {useEffect} from 'react'
+import {useHistory, useLocation} from 'react-router-dom'
+import {useQueryClient} from '@tanstack/react-query'
+import LoadingSpinner from '../loading-spinner'
+
+// For good UX, show loading spinner long enough for users to see
+const LOADING_SPINNER_MIN_DURATION = 500
+
+/**
+ * Show loading spinner while refetching data on the client-side.
+ * To trigger this refetch, we do soft navigation back to the referrer.
+ * So this component is meant to be used as a route with `referrer` search param.
+ *
+ * @example
+ * const navigate = useNavigation()
+ * navigate(`/__refetch-data?referrer=${encodeURIComponent(urlOfCurrentPage)}`, 'replace')
+ *
+ * @private
+ */
+const RefetchData = () => {
```

Comment:
naming nit: We use the URL `__mrt/refresh` but call the component `RefetchData`. Should we name it something like `Refresh` for consistency?

Reply:
Currently we assign the route to `RefetchData`, but in the future, we can have `ForceGetProps`. So the component can be swapped depending on the implementation we choose. I think the route path can be generic, while the component's name is more specific.

Regardless, since we tag the component as `@private`, we can also safely rename it later if we want to.

Reply:
Actually nvm, let me change the component name to `Refresh`.